src/main/java/org/zalando/nakadi/domain/NakadiCursor.java

```diff
@@ -1,21 +1,43 @@
 package org.zalando.nakadi.domain;
 
+import com.google.common.base.Preconditions;
 import java.util.Objects;
 
 public class NakadiCursor {
-    private final String topic;
+    public static final int VERSION_LENGTH = 3;
+    public enum Version {
+        ZERO("000"),
+        ONE("C00"),;
```

Comment:
I'm really still not sure if we should start it with "C00". Comparing BEGIN to other cursors is something wrong. Also, BEGIN is not always the smallest offset. For example when you start stream from BEGIN for some partition - this BEGIN is bigger than all offsets that are expired for this partition.